src/getConfig.js

```diff
@@ -0,0 +1,61 @@
+/* eslint no-console: 0 */
+/* eslint no-prototype-builtins: 0 */
+const intersection = require('lodash/intersection')
+const defaultsDeep = require('lodash/defaultsDeep')
+const isObject = require('lodash/isObject')
+
+/**
+ * Default config object
+ *
+ * @type {{concurrent: boolean, chunkSize: number, gitDir: string, globOptions: {matchBase: boolean, dot: boolean}, linters: {}, subTaskConcurrency: number, renderer: string, verbose: boolean}}
+ */
+const defaultConfig = {
+  concurrent: true,
+  chunkSize: Number.MAX_SAFE_INTEGER,
+  gitDir: '.',
+  globOptions: {
+    matchBase: true,
+    dot: true
+  },
+  linters: {},
+  subTaskConcurrency: 1,
+  renderer: 'update',
+  verbose: false
+}
+
+/**
+ * For a given configuration object that we retrive from .lintstagedrc or pacakge.json
+ * construct a full configuration with all options set.
+ *
+ * This is a bit tricky since we support 2 different syntxes: simple and full
+ * For simple config, only the `linters` configuration is provided.
+ *
+ * @param {Object} sourceConfig
+ * @returns {{
+ *  concurrent: boolean, chunkSize: number, gitDir: string, globOptions: {matchBase: boolean, dot: boolean}, linters: {}, subTaskConcurrency: number, renderer: string, verbose: boolean
+ * }}
+ */
+module.exports = function getConfig(sourceConfig) {
+  // Do not mutate sourceConfig!!!
+  const config = defaultsDeep({}, sourceConfig, defaultConfig)
```

Comment:
I would recommend doing this:

```js
const config = defaultsDeep(
  {},
  isSimple(sourceConfig) ? { linters: sourceConfig } : sourceConfig,
  defaultConfig
);
 
```

That way the config won't end up with garbage keys like `*.js`.